src/controllers/mainController.ts

```diff
@@ -49,6 +49,7 @@ export default class MainController implements vscode.Disposable {
     private _untitledSqlDocumentService: UntitledSqlDocumentService;
     private _objectExplorerProvider: ObjectExplorerProvider;
     private _scriptingService: ScriptingService;
+    private _connectionFromCommand: boolean = false;
```

Comment:
Having this be shared seemed dangerous - what if someone starts a connection that takes a while and then while it's spinning in the background tries to make a new connection? They'll each be overwriting this value  at different times.

Reply:
@Charles-Gagnon A node is only connected when a session is created for it. When connections are added, I have to create the session and then refresh the tree, otherwise we just get the cached disconnected nodes.

Reply:
I'm not sure how that's relevant to my concern. Even if that's the case what's stopping two nodes from being connected at nearly the same time and having the same issue?

---
In reply to: [331256285](https://github.com/microsoft/vscode-mssql/pull/1374#discussion_r331256285) [](ancestors = 331256285)